nucypher/characters/control/interfaces.py

```diff
@@ -125,6 +125,37 @@ def revoke(self, label: bytes, bob_verifying_key: bytes):
         response_data = {'failed_revocations': len(failed_revocations)}
         return response_data
 
+    def decrypt(self,
+                label: bytes,
+                policy_encrypting_key: bytes,
+                message_kit: bytes):
+        """
+        Character control endpoint to allow Alice to decrypt her own data.
+        """
+
+        from nucypher.characters.lawful import Enrico
+
+        policy_encrypting_key = UmbralPublicKey.from_bytes(
+            policy_encrypting_key)
+
+        alice_pubkey_sig = self.character.get_policy_pubkey_from_label(label)
```

Comment:
This is incorrect. `alice_pubkey_sig` should be Alice's _signing_ public key, but this line gets the policy's encrypting key. In the end, however, it seems that `alice_pubkey_sig` is not used by `alice.decrypt_message_kit()` (see comment below) so nothing actually fails. I think what you intended was:
```python
policy_encrypting_key = self.character.get_policy_pubkey_from_label(label)
```

Reply:
Ah.  Thanks!  These different keys are still a little fuzzy to me.  Will fix!